libraries/AP_HAL_ChibiOS/Util.cpp

```diff
@@ -435,20 +435,54 @@ void Util::apply_persistent_params(void) const
     char *saveptr;
     s += strlen(persistent_header);
     uint32_t count = 0;
+    uint32_t errors = 0;
     for (char *p = strtok_r(s, "\n", &saveptr);
          p; p = strtok_r(nullptr, "\n", &saveptr)) {
         char *eq = strchr(p, int('='));
         if (eq) {
             *eq = 0;
-            if (AP_Param::set_default_by_name(p, atof(eq+1))) {
+            const char *pname = p;
+            const float value = atof(eq+1);
+            if (AP_Param::set_default_by_name(pname, value)) {
                 count++;
+                /*
+                  we now have a special case for INS_ACC*_ID. To
+                  support factory accelerometer calibration we need to
+                  do a save() on the ID parameters if they are not
+                  already in storage. This is needed as
+                  AP_InertialSensor determines if a calibration has
+                  been done by whether the IDs are configured in
+                  storage
+                 */
+                if (strncmp(pname, "INS_ACC", 7) == 0 &&
```

Comment:
Presumably the way you have organized storage it doesn't matter if the parameter name is 16 chars with no trailing 0?

Reply:
storage is as strings, just like a param file